Tests/TuistCoreTests/ValueGraph/ValueGraphTraverserTests.swift

```diff
@@ -1525,6 +1525,58 @@ final class ValueGraphTraverserTests: TuistUnitTestCase {
         XCTAssertEqual(got, gotGraph)
         XCTAssertEqual(gotGraph.first, GraphDependencyReference(precompiledNode: precompiledNode))
     }
+    
+    func test_linkableAndEmbeddableDependencies_when_appDependensOnPrecompiledStaticBinaryWithPrecompiledStaticBinaryDependency() throws {
```

Comment:
I'd like to kindly ask you @kwridan to check if the "Given: Value Graph" part of a unit test is correct 🙏  Please note that this test is failing, because a value graph linkable output is not equal to graph traverser output. 

```
XCTAssertEqual(got, gotGraph)
```

I haven't made the changes in the `ValueGraph` yet, however, I am wondering if maybe we should deliver it in the separate pull request due to taking time to figure out the best solution and the fact that we already have pretty decent solution for a reference graph. What do you think? CC: @pepibumur